### PR TITLE
Improve the output of the commands

### PR DESCRIPTION
**Pull request by MKodde**
So `bin/console user-lifecycle:information --user=foobar` turns into
`bin/console user-lifecycle:information foobar`

### CODE REVIEW

File: src/OpenConext/UserLifecycle/Application/Service/DeprovisionService.php [hunk 1]
```diff
@@ -95,7 +97,7 @@ public function deprovision($personId, $dryRun = false)
 
         $this->logger->info(
             sprintf('Received deprovision information for user "%s" with the following data.', $personId),
-            ['information_response' => $information->jsonSerialize()]
+            ['information_response' => json_encode($information->jsonSerialize())]
```
Comment (jorissteyn) [inline src/OpenConext/UserLifecycle/Application/Service/DeprovisionService.php]: json_encode($information) would do

File: src/OpenConext/UserLifecycle/Application/Service/SummaryService.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        $errorMessageList = '';
+        if (!empty($errorMessages)) {
+            $errorMessageList .= sprintf(self::USER_DEPROVISION_ERROR_FORMAT) . PHP_EOL . PHP_EOL;
+
+            foreach ($errorMessages as $errorMessage) {
+                $errorMessageList .= ' * ' . $errorMessage . PHP_EOL;
+            }
+        }
+
+        return $message . $errorMessageList;
```
Comment (jorissteyn) [inline src/OpenConext/UserLifecycle/Application/Service/SummaryService.php]: Idea: why not create a list of lines, and return implode(PHP_EOL, $lines).

File: src/OpenConext/UserLifecycle/Application/Service/SummaryService.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    /**
+     * Set the context to deprovision or information outputting
+     *
+     * @param string $context
+     */
+    public function setContext($context)
```
Comment (jorissteyn) [inline src/OpenConext/UserLifecycle/Application/Service/SummaryService.php]: Would be nicer if this object was stateless, you could remove the first line in `summarizeInformationResponse()` and allow users (of the class) to call a new public function something along the lines of `printSummaryHeaderForInformationRequest()` and `printSummaryHeaderForDeprovisionRequest()`.

File: src/OpenConext/UserLifecycle/Domain/Client/BatchInformationResponseCollectionInterface.php [hunk 4]
```diff
@@ -33,8 +33,19 @@
      */
     public function add(CollabPersonId $personId, InformationResponseCollectionInterface $collection);
 
+    /**
+     * @return int
+     */
+    public function count();
```
Comment (jorissteyn) [inline src/OpenConext/UserLifecycle/Domain/Client/BatchInformationResponseCollectionInterface.php:40]: Could just as well implement Countable.

File: src/OpenConext/UserLifecycle/Infrastructure/UserLifecycleBundle/Command/DeprovisionCommand.php [hunk 5]
```diff
@@ -139,7 +175,15 @@ private function executeSingleUser(InputInterface $input, OutputInterface $outpu
...
-            $output->write($this->service->deprovision($userIdInput, $dryRun));
+            $information = $this->service->deprovision($userIdInput, $dryRun);
+
+            if (!$isQuiet) {
+                $output->writeln(PHP_EOL);
+                $output->write($this->summaryService->summarize($information), true);
+                $output->writeln('Full output of the deprovisioning command:' . PHP_EOL);
+            }
+
+            $output->write(json_encode($information->jsonSerialize()), true);
```
Comment (jorissteyn) [inline src/OpenConext/UserLifecycle/Infrastructure/UserLifecycleBundle/Command/DeprovisionCommand.php]: See comment above about `jsonSerialize()`, goes for other occurrences too.

File: tests/unit/Application/Service/SummaryServiceTest.php [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                '"EngineBlock not available" for user "urn:collab:jane_doe"',
+                '"DropjesService has gone away" for user "urn:collab:jack_black"',
+            ]);
+
+        $summary = $this->service->summarize($collection);
+
+        $this->assertContains('256 users have been deprovisioned.', $summary);
+        $this->assertContains('3 deprovision calls to services failed. See error messages below:', $summary);
+        $this->assertContains(' * "Error message" for user "collabPersonId"', $summary);
+        $this->assertContains(' * "EngineBlock not available" for user "urn:collab:jane_doe"', $summary);
```
Comment (jorissteyn) [inline tests/unit/Application/Service/SummaryServiceTest.php]: In practive, the message probably won't be "EngineBlock not available" but something like "Lorem ipsum integrity constraint failure". What I mean is: shouldn't we add the name of the failing application.

File: src/OpenConext/UserLifecycle/Application/Service/DeprovisionService.php (hunk 1 above)
Comment (MKodde) [reply src/OpenConext/UserLifecycle/Application/Service/DeprovisionService.php]: d'oh

File: tests/unit/Application/Service/SummaryServiceTest.php (hunk 6 above)
Comment (MKodde) [reply tests/unit/Application/Service/SummaryServiceTest.php]: Yes, good point!

**Pull request merged**